### PR TITLE
perf(engine): speed up piece-property validation by removing per-call schema overhead

### PR DESCRIPTION
**Pull request by abuaboud**
## Summary

Speeds up every piece-step execution that validates inputs (which is every piece step) by removing per-call schema-object construction from the property validator.

## Why

`packages/server/engine/src/lib/variables/props-processor.ts:validateProperty` ran once per property per step and recursed into ARRAY items and DYNAMIC sub-props. Each invocation built a fresh `z.string()` / `z.number()` / `z.array(...)` / `z.record(...)` / `.refine(...)` / `.nullable().optional()` chain.

Zod 4's `$constructor.init` allocates a bound-function closure for every prototype method on every schema instance:

```js
for (let i = 0; i < keys.length; i++) {
  const k = keys[i];
  if (!(k in inst)) {
    inst[k] = proto[k].bind(inst);
  }
}
```

A `ZodString` carries ~15-20 prototype methods, so each `z.string()` allocated ~15-20 closures plus a `Set` plus a `_zod` metadata object. Chained variants tripled that. A piece step with an ARRAY of N items × M sub-properties built O(N·M) schemas per execution — for large inputs that's hundreds of thousands of short-lived closures spinning up the GC and inflating engine-subprocess heap.

`validateProperty` only ever asks "is this a string / number / boolean / array / object / JSON?" That's a fixed shape check that costs zero allocations when done directly.

## What changed

`packages/server/engine/src/lib/variables/props-processor.ts`:

- Dropped the `zod` import.
- `validateProperty` is now a switch over `typeof` / `Array.isArray` / `isObject` returning the same error strings.
- JSON is handled at the top of the function so its `originalValue`-vs-`value` inspection (the "user provided something but it didn't resolve" branch) fires before the shared `!required && isNil(value)` bailout that mirrors `schema.nullable().optional()`.
- NUMBER check is `typeof === 'number' && !Number.isNaN(value)` to match Zod 4's `z.number()` semantics — the number processor coerces via `Number(value)` and can return `NaN`.

The agent-tools file (`tools/index.ts`) is intentionally **not** touched in this PR — that change can land separately.

## Test plan

- [x] `packages/server/engine/test/variables/props-validator.test.ts` — 9/9 pass (covers required + optional × all property types, including JSON edge cases and NaN-from-coercion).
- [x] `npm run test-unit` — engine + shared + web all green.
- [x] `npx turbo run lint --filter=@activepieces/engine` — 0 errors.
- [ ] Manual: run a piece step with an ARRAY input + nested properties; confirm validation errors match prior behavior.

### CODE REVIEW

File: packages/server/engine/src/lib/tools/index.ts [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import { Action, DropdownOption, ExecutePropsResult, PieceProperty, PropertyType } from '@activepieces/pieces-framework'
```
[the remark this answers is not in the document]
Comment (abuaboud) [reply packages/server/engine/src/lib/tools/index.ts:1]: Title updated to "perf(engine): speed up piece-property validation by removing per-call schema overhead" — leads with user-visible impact (piece-property validation speed) instead of the internal change (dropping Zod). Description rewritten in the same direction. Resolving.

File: packages/server/engine/src/lib/tools/index.ts [hunk 2]
```diff
@@ -256,71 +258,68 @@ async function propertyToSchema(propertyName: string, property: PieceProperty, o
...
         case PropertyType.SECRET_TEXT:
             throw new Error(`Unsupported property type: ${property.type}`)
     }
+    const wrapped: JSONSchema7 = property.required ? { ...schema } : { anyOf: [schema, { type: 'null' }] }
     if (property.description) {
-        schema = schema.describe(property.description)
+        wrapped.description = property.description
     }
-    return property.required ? schema : schema.nullable()
+    return wrapped
```
[the remark this answers is not in the document]
Comment (abuaboud) [reply packages/server/engine/src/lib/tools/index.ts]: No longer applies — the agent-tools change in `tools/index.ts` was reverted; this PR's scope is now only `props-processor.ts`. The nested-`anyOf` concern only lived in the JSON Schema replacement on the tools file. Resolving.
